teachertool/src/transforms/runEvaluateAsync.ts

```diff
@@ -29,7 +33,42 @@ function generateValidatorPlan(criteriaInstance: CriteriaInstance): pxt.blocks.V
         return undefined;
     }
 
-    // TODO: Fill in any parameters. Error if parameters are missing.
+    // Fill in parameters.
+    for (const param of criteriaInstance.params ?? []) {
+        const catalogParam = catalogCriteria.params?.find(p => p.name === param.name);
+        if (!catalogParam) {
+            if (showErrors) {
+                logError(
+                    ErrorCode.evalMissingCatalogParameter,
+                    "Attempting to evaluate criteria with unrecognized parameter",
+                    { catalogId: criteriaInstance.catalogCriteriaId, paramName: param.name }
+                );
+            }
+            return undefined;
+        }
+
+        if (!param.value) {
+            // User didn't set a value for the parameter.
+            if (showErrors) {
+                logError(ErrorCode.evalParameterUnset, "Attempting to evaluate criteria with unset parameter value", {
+                    catalogId: criteriaInstance.catalogCriteriaId,
+                    paramName: param.name,
+                });
+                showToast(
+                    makeToast(
+                        "error",
+                        // prettier-ignore
+                        lf("Unable to evaluate criteria: missing '{0}' in '{1}'", param.name, getReadableCriteriaTemplate(catalogCriteria))
+                    )
+                );
+            }
+            return undefined;
```

Comment:
What would happen if we don't return here or above when we get an error? Would it show all of the problems with that one criteria if there are multiple? Would it be too overwhelming for the user?

Just curious about these things, not saying that we need to change anything here.

Reply:
As implemented currently, it would actually end up running the check, but it'd use the values we have stored in the validatorPlan pre-replacement (often empty strings or 0), so it would look like the check ran but what it really checked would be something different.

We could expand this to have a more detailed validation where we parse the whole thing in advance and make a toast with all errors combined, but I'm not sure that's worth the trouble. Ideally any errors would be immediately apparent from the rubric display regardless, though I'm planning to do that in a follow-up change.